src/azure-cli/azure/cli/command_modules/backup/custom_wl.py

```diff
@@ -0,0 +1,584 @@
+
+
+def list_wl_policies(client, resource_group_name, vault_name, workload_type, backup_management_type):
+    if workload_type is None:
+        raise CLIError(
+            """
+            Workload type is required for Azure Workload.
+            """)
+
+    if backup_management_type is None:
+        raise CLIError(
+            """
+            Backup Management Type needs to be specified for Azure Workload.
+            """)
+
+    workload_type = workload_type_map[workload_type]
+
+    filter_string = cust_help.get_filter_string({
+        'backupManagementType': backup_management_type,
+        'workloadType': workload_type})
+
+    policies = client.list(vault_name, resource_group_name, filter_string)
+    return cust_help.get_list_from_paged_response(policies)
+
+
+def list_protectable_containers(cmd, resource_group_name, vault_name, container_type="AzureWorkload"):
+    filter_string = cust_help.get_filter_string({
+        'backupManagementType': container_type})
+    client = protectable_containers_cf(cmd.cli_ctx)
+    paged_containers = client.list(vault_name, resource_group_name, fabric_name, filter_string)
+    return cust_help.get_list_from_paged_response(paged_containers)
+
+
+def register_wl_container(cmd, client, vault_name, resource_group_name, workload_type, resource_id, container_type):
+    if not cust_help.is_id(resource_id):
+        raise CLIError(
+            """
+            Resource ID is not a valid one.
+            """)
+
+    workload_type = workload_type_map[workload_type]
+    container_name = resource_id.split('/')[-1]
+
+    containers = list_protectable_containers(cmd, resource_group_name, vault_name)
+
+    for container in containers:
+        if cust_help.get_resource_id(container.properties.container_id) == cust_help.get_resource_id(resource_id):
+            container_name = container.name
+            break
+
+    if not cust_help.is_native_name(container_name):
+        raise CLIError(
+            """
+            Container unavailable or already registered.
+            """)
+
+    properties = AzureVMAppContainerProtectionContainer(backup_management_type=container_type,
+                                                        source_resource_id=resource_id,
+                                                        workload_type=workload_type)
+    param = ProtectionContainerResource(properties=properties)
+
+    # Trigger register and wait for completion
+    result = sdk_no_wait(True, client.register,
+                         vault_name, resource_group_name, fabric_name, container_name, param)
+    return cust_help.track_register_operation(cmd.cli_ctx, result, vault_name, resource_group_name, container_name)
+
+
+def re_register_wl_container(cmd, client, vault_name, resource_group_name, workload_type,
+                             container_name, container_type):
+    workload_type = workload_type_map[workload_type]
+
+    if not cust_help.is_native_name(container_name):
```

Comment:
It looks like a validation function and is used in register https://github.com/Azure/azure-cli/pull/11066/files#diff-afdaa9416bdf880925ee9d3df0e072baR103. Can we consider abstract it and add as validator kwargs?

Reply:
- This function is only used as a validator in unregister command. 
- In other commands, it is used to check if service is returning any containers or not.
- container_name is a common parameter used by many commands, most of which don't require this validation. So i think it'd be better not to abstract this function just for one command.